src/clothing_mod.h

```diff
@@ -45,14 +49,84 @@ struct clothing_mod {
     float get_mod_val( const clothing_mod_type &type, const item &it ) const;
     bool has_mod_type( const clothing_mod_type &type ) const;
 
+    /**
+     * Check whether the mod applies a given flag.
+     *
+     * @param f a flag string to check
+     * @return bool true if the mod would apply the flag
+     */
+    bool applies_flag( const std::string &f ) const;
+
+    /**
+     * Check whether the mod suppresses a given flag.
+     *
+     * @param f a flag string to check
+     * @return bool true if the mod would suppress the flag
+     */
+    bool suppresses_flag( const std::string &f ) const;
+
+    /**
+     * Whether the mod applies any flags when used.
+     *
+     * @return bool true if the mod applies flags
+     */
+    bool applies_flags( ) const;
```

Comment:
nit: these shouldn't have spaces (though i guess astyle doesn't enforce this)
```suggestion
    bool applies_flags() const;
```